scripts/mainfork-snapshot.sh

```diff
@@ -0,0 +1,275 @@
+#! /usr/bin/env bash
+
+ACCESS_TOKEN_URI="https://oauth2.googleapis.com/token"
+BUCKET_NAME="agoric-snapshots-public"
+CONTAINER_MEMORY="200GiB"
+IMAGE_TAG="${IMAGE_TAG:-"57"}"
+FIRST_NODE_DATA_FOLDER_NAME="agoric1"
+FIRST_NODE_IP="10.99.0.2"
+PARENT_FOLDER="/var/lib/google/code"
+PROJECT_NAME="simulationlab"
+REGION="us-central1-a"
+REPOSITORY_URL="https://github.com/agoric-labs/cosmos-genesis-tinkerer.git"
+STORAGE_WRITE_SCOPES="https://www.googleapis.com/auth/devstorage.read_write"
+SECOND_NODE_DATA_FOLDER_NAME="agoric2"
+SECOND_NODE_IP="10.99.0.3"
+STORAGE_UPLOAD_URL="https://storage.googleapis.com/upload/storage/v1/b"
+TIMESTAMP="$(date '+%s')"
+VALIDATOR_STATE_FILE_NAME="priv_validator_state.json"
+VM_NAME="jump-3"
+VM_OPERATIONS_SCOPES="https://www.googleapis.com/auth/cloud-platform"
+VM_RUNNING_STATUS="RUNNING"
+VM_STOPPED_STATUS="TERMINATED"
+
+FIRST_NODE_LOGS_FILE="/tmp/$FIRST_NODE_DATA_FOLDER_NAME.logs"
+IMAGE_NAME="ghcr.io/agoric/agoric-sdk:$IMAGE_TAG"
+LOGS_FILE="/tmp/$TIMESTAMP.logs"
+REPOSITORY_FOLDER_NAME="tinkerer_$TIMESTAMP"
+SECOND_NODE_LOGS_FILE="/tmp/$SECOND_NODE_DATA_FOLDER_NAME.logs"
+STORAGE_WRITE_SERVICE_ACCOUNT_JSON_FILE_PATH="$PARENT_FOLDER/chain-snapshot-writer.json"
+VM_ADMIN_SERVICE_ACCOUNT_JSON_FILE_PATH="$PARENT_FOLDER/vm-admin.json"
+
+execute_command_inside_vm() {
+    local command="$1"
+    gcloud compute ssh "$VM_NAME" \
+        --command "$command" --project "$PROJECT_NAME" --zone "$REGION"
+}
+
+get_vm_status() {
+    gcloud compute instances describe "$VM_NAME" \
+        --format "value(status)" --project "$PROJECT_NAME" --zone "$REGION"
+}
+
+log_warning() {
+    printf "\033[33m%s\033[0m\n" "$1"
+}
+
+signal_vm_start() {
+    gcloud compute instances start "$VM_NAME" \
+        --async --project "$PROJECT_NAME" --zone "$REGION" >/dev/null 2>&1
+}
+
+start_vm() {
+    if test "$(get_vm_status)" = "$VM_STOPPED_STATUS"; then
+        log_warning "Starting VM $VM_NAME"
+        signal_vm_start
+        wait_for_vm_status "$VM_RUNNING_STATUS"
+    else
+        log_warning "VM $VM_NAME already running"
+    fi
+}
+
+wait_for_vm_status() {
+    local status="$1"
+    while [ "$(get_vm_status)" != "$status" ]; do
+        sleep 5
+    done
+}
+
+start_vm
+execute_command_inside_vm "
+    #! /bin/bash
+
+    set -o errexit -o errtrace
```

Comment:
I'm a little worried this could publicly print out access tokens to our Google Cloud environment. What restrictions / expirations exist on those?